nistats/tests/test_datasets.py

```diff
@@ -174,18 +175,15 @@ def _expected_output_data_from_test_file():
     def run_test():
         data_for_tests = _input_data_for_test_file()
         expected_data_from_test_file = _expected_output_data_from_test_file()
-        with NamedTemporaryFile(mode='w',
-                                dir=os.getcwd(),
-                                suffix='.csv') as temp_csv_obj:
-            data_for_tests.to_csv(temp_csv_obj.name,
+        temp_tsv_file = 'temp_file.tsv'
+        with InTemporaryDirectory():
```

Comment:
You are not following my suggestion, what's wrong with it ?

Reply:
I had changed temp_csv_obj.name with temp_csv_obj as you suggested and it was working fine. However, an error unrelated to that happened that it was not reading the csv properly into a pandas dataframe. Skipping NamedTemporaryFile easily solved this with a one line change..